pkg/preflight/checks.go

```diff
@@ -0,0 +1,138 @@
+package preflight
+
+import (
+	"bytes"
+	"fmt"
+	"log"
+	"net"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+
+	api "github.com/weaveworks/ignite/pkg/apis/ignite"
+	"github.com/weaveworks/ignite/pkg/constants"
+	"github.com/weaveworks/ignite/pkg/operations/lookup"
+	"github.com/weaveworks/ignite/pkg/providers"
+)
+
+const (
+	oldPathString  = "/"
+	newPathString  = "-"
+	noReplaceLimit = -1
+)
+
+type Checker interface {
+	Check() error
+	Name() string
+	Type() string
+}
+
+type PortOpenChecker struct {
+	port uint64
+}
+
+func (poc PortOpenChecker) Check() error {
+	listener, err := net.Listen("tcp", fmt.Sprintf(":%d", poc.port))
+	if err != nil {
+		return fmt.Errorf("Port %d is in use", poc.port)
+	}
+	if err := listener.Close(); err != nil {
+		return fmt.Errorf("Port %d is in used, failed to close it", poc.port)
+	}
+	return nil
+}
+
+func (poc PortOpenChecker) Name() string {
+	return fmt.Sprintf("Port-%d", poc.port)
+}
+
+func (poc PortOpenChecker) Type() string {
+	return "Port"
+}
+
+type ExistingFileChecker struct {
+	filePath string
+}
+
+func (efc ExistingFileChecker) Check() error {
+	if _, err := os.Stat(efc.filePath); os.IsNotExist(err) {
+		return fmt.Errorf("File %s, does not exist", efc.filePath)
+	}
+	return nil
+}
+
+func (efc ExistingFileChecker) Name() string {
+	return fmt.Sprintf("ExistingFile-%s", strings.Replace(efc.filePath, oldPathString, newPathString, noReplaceLimit))
+}
+
+func (efc ExistingFileChecker) Type() string {
+	return "ExistingFile"
+}
+
+type AvailablePathChecker struct {
+	path string
+}
+
+func (apc AvailablePathChecker) Check() error {
+	if _, err := os.Stat(apc.path); !os.IsExist(err) {
+		return fmt.Errorf("Path %s, already exist", apc.path)
+	}
+	return nil
+}
+
+func (apc AvailablePathChecker) Name() string {
+	return fmt.Sprintf("AvailablePath-%s", strings.Replace(apc.path, oldPathString, newPathString, noReplaceLimit))
+}
+
+func (apc AvailablePathChecker) Type() string {
+	return "AvailablePath"
+}
+
+func StartCmdChecks(vm *api.VM, ignoredPreflightErrors sets.String) error {
+	checks := []Checker{}
+	kernelUID, err := lookup.KernelUIDForVM(vm, providers.Client)
+	if err != nil {
+		return err
+	}
+	vmDir := filepath.Join(constants.VM_DIR, vm.GetUID().String())
+	kernelDir := filepath.Join(constants.KERNEL_DIR, kernelUID.String())
+	log.Println(vm.GetUID().String())
+	checks = append(checks, ExistingFileChecker{filePath: path.Join(vmDir, constants.METADATA)})
```

Comment:
No need to check for the metadata and kernel files, they are gonna be created on the fly.